adapter/operations.go

```diff
@@ -14,15 +14,46 @@
 
 package adapter
 
-import "context"
+import (
+	"context"
+	"net/url"
 
-const (
-	OperationDescriptionKey  = "description"
-	OperationVersionKey      = "version"
-	OperationTemplateNameKey = "templateName"
-	OperationServiceNameKey  = "serviceName"
+	"github.com/layer5io/meshkit/utils"
 )
 
+var (
+	NoneVersion  = []Version{"none"}
+	NoneTemplate = []Template{"none"}
+)
+
+type Version string
+
+type Template string
+
+func (t Template) String() string {
+	_, err := url.ParseRequestURI(string(t))
+	if err != nil {
+		return string(t)
+	}
+
+	st, err := utils.ReadRemoteFile(string(t))
```

Comment:
@kumarabd Maybe change the name of this func, to `ReadResource`? It can also be local, need not be a file, and we treat it as a resource. Could be `file://`, `http://`, `gs://`, a document in a document database etc.